File: 0014_longest-common-prefix/csharp/solution.cs
public class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        if (strs.Length is 0)
            return "";

        var reference = strs[0];

        for (int i = 0; i < reference.Length; i++)
        {
            var c = reference[i];
            for (int j = 1; j < strs.Length; j++)
            {
                if (i >= strs[j].Length || strs[j][i] != c)
                    return reference[0..i];
            }
        }

        return reference;
    }
}

File: 0014_longest-common-prefix/rust/solution.rs
#[cfg(feature = "harness")]
pub struct Solution;

impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() {
            return "".into();
        }

        let rf = &strs[0];

        for i in 0..rf.len() {
            let c = rf.as_bytes()[i];
            let iter = 1..strs.len();
            for j in iter {
                let curr = &strs[j];
                if i >= curr.len() || curr.as_bytes()[i] != c {
                    return rf[0..i].into();
                }
            }
        }
        rf.into()
    }
}
